jumpscale/packages/vdc_dashboard/chats/matic.py

```diff
@@ -94,17 +97,35 @@ def _enter_sentry_info(self):
         self.config.chart_config.extra_config["sentry_nodeid"] = sentry_nodeid.value
         self.config.chart_config.extra_config["sentry_enodeid"] = sentry_enodeid.value
 
+    @chatflow_step(title="Web Access")
     def _enter_access_code(self):
         self.config.chart_config.access_code = self.secret_ask(
             "Enter Access Code (This would be used to access your node's web page)", min_length=8, required=True
         )
 
-    @chatflow_step(title="Node Configuration")
+    @chatflow_step(title="Port Settings")
+    def _get_node_ports(self):
+        form = self.new_form()
+        heimdall_port = form.int_ask("Heimdall service port - default is 26656", default=26656, min=1000, max=65000)
```

Comment:
if that's the node ports service, then the range needs to be similar to what is configured in service-cluster-ip-range, which is most likey to be ~30000 and ~32000, and I don't see any logic handling the wrong choice of the ports or if it already used by something else.

Reply:
We are requesting the user to use different ports. If you have a code snippet / example that can checks the ports usage of deployed solutions, please share

Reply:
This has been resolved by adding a [tcp ingress helper function](https://github.com/threefoldtech/js-sdk/pull/3082/files#diff-71171f1090c4eb9dbd3a3af14860fe6f73496fd937480e2dae6238ef288c91deR246) that returns tcp ports in use 

![image](https://user-images.githubusercontent.com/25789764/118590229-112b2000-b7b3-11eb-99f3-5147e0ecdeb4.png)